File: test_verify_auth_state.py
from playwright.sync_api import sync_playwright
import time

def test_flow():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()

        timestamp = int(time.time() * 1000)
        test_email = f"test_{timestamp}@example.com"

        page.goto("http://localhost:4200/register")
        page.get_by_label("Adresse email principale").fill(test_email)
        page.get_by_label("Mot de passe maitre").fill("password123")
        page.get_by_role("button", name="S'inscrire").click()

        page.wait_for_url("http://localhost:4200/inbox", timeout=10000)

        page.get_by_role("button", name="Parametres").click()
        page.get_by_text("General").click()
        page.locator('select#undo-delay').select_option(label="5 secondes")
        page.wait_for_timeout(2000)

        profile = page.evaluate("""async () => {
            const token = localStorage.getItem('auth_token');
            const res = await fetch('http://localhost:3300/api/auth/profile', {
                headers: { 'Authorization': 'Bearer ' + token }
            });
            return await res.json();
        }""")
        print("Profile after setting:", profile)
        browser.close()

if __name__ == "__main__":
    test_flow()

File: test_verify_auth_state.js
const { chromium } = require('playwright');
(async () => {
    const browser = await chromium.launch();
    const page = await browser.newPage();

    await page.goto("http://localhost:4200/register");
    const test_email = `test_${Date.now()}@example.com`;
    await page.fill('label:has-text("Adresse email principale") + input', test_email);
    await page.fill('label:has-text("Mot de passe maitre") + input', 'password123');
    await page.click('button:has-text("S\'inscrire")');

    await page.waitForURL('http://localhost:4200/inbox');

    // Check initial user state via localStorage token + api or angular debug

    await page.click('button[aria-label="Parametres"]');
    await page.click('button:has-text("General")');
    await page.selectOption('select#undo-delay', { label: '5 secondes' });
    await page.waitForTimeout(1000);

    // Request profile to see if delay is saved
    const response = await page.evaluate(async () => {
        const token = localStorage.getItem('auth_token');
        const res = await fetch('http://localhost:3300/api/auth/profile', {
            headers: { 'Authorization': `Bearer ${token}` }
        });
        return await res.json();
    });
    console.log("Profile after setting:", response);

    await browser.close();
})();
